### PR TITLE
Add minimal post-fitering of the BLAST results

### PR DESCRIPTION
**Pull request by laughedelic**
Minimal implementation of things described in #38

### CODE REVIEW

**Comment by laughedelic** [comment]
@eparejatobes review please

**Comment by eparejatobes** [comment]
:+1:

File: src/main/scala/metagenomica/parameters.scala [hunk 1]
```diff
@@ -38,33 +39,66 @@ trait AnyMG7Parameters {
...
+  implicit val has_qcovs:    out.qcovs.type    isOneOf BlastOutRecKeys#Types#AllTypes
+
+  // NOTE: this is not exposed among other constructor arguments, but you can _override_ it
+  val blastFilter: csv.Row[BlastOutRecKeys] => Boolean = defaultBlastFilter
+
+  // NOTE: this default is defined here to have has_qcovs implicit in the scope
+  val defaultBlastFilter: csv.Row[BlastOutRecKeys] => Boolean = { row =>
+
+    val qcovs: String = row.select(outputFields.qcovs)
+    parseDouble(qcovs).map(_ > 99.5).getOrElse(false)
```
Comment (laughedelic) [inline src/main/scala/metagenomica/parameters.scala]: @eparejatobes @rtobes should it rather be `100`?

File: src/main/scala/metagenomica/parameters.scala [hunk 2]
```diff
@@ -38,33 +39,66 @@ trait AnyMG7Parameters {
...
-  val argValsToSeq: BlastOptionsToSeq[BlastArgumentsVals] = implicitly[BlastOptionsToSeq[BlastArgumentsVals]]
-  val optValsToSeq: BlastOptionsToSeq[BlastCommand#OptionsVals] // has to be provided implicitly
+  implicit val argValsToSeq: BlastOptionsToSeq[BlastArgumentsVals] = implicitly[BlastOptionsToSeq[BlastArgumentsVals]]
+  implicit val optValsToSeq: BlastOptionsToSeq[BlastCommand#OptionsVals] // has to be provided implicitly
+
+  // The minimal set of the output fields neccessary for the MG7 generic code
+  implicit val has_qseqid:   out.qseqid.type   isOneOf BlastOutRecKeys#Types#AllTypes
+  implicit val has_sseqid:   out.sseqid.type   isOneOf BlastOutRecKeys#Types#AllTypes
+  implicit val has_bitscore: out.bitscore.type isOneOf BlastOutRecKeys#Types#AllTypes
+  implicit val has_pident:   out.pident.type   isOneOf BlastOutRecKeys#Types#AllTypes
```
Comment (laughedelic) [inline src/main/scala/metagenomica/parameters.scala]: @eparejatobes if we can set `perc_identity` as an option, is this output field still necessary? (it's not used anywhere in the code, unlike the others)

Comment (eparejatobes) [inline src/main/scala/metagenomica/parameters.scala]: I would not drop it

File: src/main/scala/metagenomica/parameters.scala (hunk 1 above)
Comment (eparejatobes) [inline src/main/scala/metagenomica/parameters.scala]: Let's see. I would start with 98 or something similar.

**Pull request merged**